server/backend/database/memory/database.go

```diff
@@ -243,12 +244,47 @@ func (d *DB) listAllProjectInfos(
 	}
 
 	var infos []*database.ProjectInfo
+
 	for raw := iter.Next(); raw != nil; raw = iter.Next() {
 		info := raw.(*database.ProjectInfo).DeepCopy()
 		infos = append(infos, info)
 	}
 
-	return infos, nil
+	sort.SliceStable(infos, func(i, j int) bool {
+		firstIDBytes, err := infos[i].ID.Bytes()
+		if err != nil {
+			return true
+		}
+		secondIDBytes, err := infos[j].ID.Bytes()
+		if err != nil {
+			return true
+		}
+
+		return bytes.Compare(firstIDBytes, secondIDBytes) < 0
+	})
+
+	beginIndex := 0
+	for i := 0; i < len(infos); i++ {
+		proejctIDBytes, err := infos[i].ID.Bytes()
+		if err != nil {
+			return nil, fmt.Errorf("decode project id: %w", err)
+		}
+		lastProjectIDBytes, err := housekeepingLastProjectID.Bytes()
+		if err != nil {
+			return nil, fmt.Errorf("decode last project id: %w", err)
+		}
+		if bytes.Compare(proejctIDBytes, lastProjectIDBytes) > 0 {
+			beginIndex = i
+			break
+		}
+	}
+
+	lastIndex := beginIndex + pageSize - 1
+	if lastIndex >= len(infos) {
+		lastIndex = len(infos) - 1
+	}
+
+	return infos[beginIndex : lastIndex+1], nil
```

Comment:
How about using an index to avoid fetching all projects from DB? MemDB also provides indexes like other DB.

You can refer to the below about how to use indexes in MemDB:
- https://github.com/yorkie-team/yorkie/blob/16fd182021231d75562a933cb32d924af16fc7f4/server/backend/database/memory/database.go#L913-L942
- https://github.com/hashicorp/go-memdb#example

Reply:
Thank you for the feedback. I didn't notice that in MemDB. I added changes to use an index.